internal/githubapi/errors.go

```diff
@@ -95,11 +95,7 @@ func (e *GraphQLErrors) Errors() []GraphQLError {
 
 // Is implements the errors.Is interface.
 func (e *GraphQLErrors) Is(target error) bool {
-	if target == e {
-		// Identity test is always true.
-		return true
-	}
-	if target == ErrGraphQLNotFound {
```

Comment:
The correct way to compare errors is `errors.Is()`

Reply:
That is true generally, however, in this case we are implementing the `Is(error) errror` interface that makes `errors.Is` work.

Comparing directly using `target == ErrGraphQLNotFound` is the right way to do this.

The docs on https://pkg.go.dev/errors#Is say `"An Is method should only shallowly compare err and the target and not call Unwrap on either."` - calling `errors.Is()` in this context would cause an `Unwrap` to occur violating this.

The docs even show a similar example: `target == fs.ErrExist`.

Reply:
Agreed, thanks.